docker-compose.no-db.yml

```diff
@@ -0,0 +1,38 @@
+services:
+  fidesops:
+    container_name: fidesops
+    build:
+      context: .
+      dockerfile: Dockerfile
+    expose:
+      - 8080
+    healthcheck:
+      test: [ "CMD", "curl", "-f", "http://0.0.0.0:8080/health" ]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 1s
+    ports:
+      - "8080:8080"
+    volumes:
+      - type: bind
+        source: ./
+        target: /fidesops
+        read_only: False
+      - /fidesops/src/fidesops.egg-info
+    environment:
+      - FIDESOPS__DEV_MODE=${FIDESOPS__DEV_MODE}
+      - FIDESOPS__LOG_PII=${FIDESOPS__LOG_PII}
+      - FIDESOPS__HOT_RELOAD=${FIDESOPS__HOT_RELOAD}
+      - FIDESOPS__DATABASE__ENABLED=false
+      - FIDESOPS__REDIS__ENABLED=false
```

Comment:
Commenting here with some `docker compose` trivia... you can avoid this separate `no-db.yml` by exposing this ENV variable on the main file like so:
```
- FIDESOPS__DATABASE__ENABLED =${FIDESOPS__DATABASE__ENABLED:-true}
```
That will default to true, or take the value of the ENV var provided to the docker compose invocation.

Then you just need to set the value of the ENV variable in the Makefile to adjust that behaviour, like:
```
server-no-db: export FIDESOPS__DATABASE__ENABLED=false
server-no-db: ... <continue existing target here>
```

I do this exact pattern in fidesdemo, see:
* https://github.com/ethyca/fidesdemo/blob/main/docker-compose.yml#L68
* https://github.com/ethyca/fidesdemo/blob/main/Makefile#L189

Reply:
certainly more concise for the ENV var piece of it, but the `no-db.yml` compose also doesn't bootstrap the `db` and `redis` (and `docs`) services, to more closely simulate the "standalone" deployment. as far as i know (and based on our conversation on friday), i don't think there's a way to have branching logic in `docker-compose` based on env var? but i can dig on that if you think it's possible.

Reply:
Ah yes, that's true... it's Hard™️ to make docker compose files dynamic and support different run configurations.

Reply:
TIL this exists: https://docs.docker.com/compose/profiles/

Reply:
oh - that looks pretty nice. i can take a shot at that, seems a lot cleaner. thanks for pointing me to that!

Reply:
hmm. because we've got dependencies between `fidesops` and `db` and `redis`, i don't think this is going to work for us, without some significant refactor. as far as i can tell, there's no way to express conditional dependencies, and i feel like the no-db use case is enough of an edge case for now that i am reluctant to remove those dependencies...